CMakeLists.txt

```diff
@@ -102,6 +102,20 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     )
 endif()
 
+if(CMAKE_COMPILE_WARNING_AS_ERROR)
+    # These -Wno-error flags are here to enable -Werror for clazy in CI
+    # it doesn't mean we endorse those warnings.
+    # MSVC Werror is untested and probably will have a few hundered errors
+    if(MSVC)
+        add_compile_options(/WX)
```

Comment:
This can be removed the cmake option takes care of that

Modules/Compiler/MSVC.cmake:13:   set(CMAKE_${lang}_COMPILE_OPTIONS_WARNING_AS_ERROR "-WX")